post-scan-actions/aws-python-promote-or-quarantine/handler.py

```diff
@@ -21,22 +37,37 @@ def parse_s3_object_url(url_string):
 
     return bucket, object_key
 
-def copy_object(source_bucket, source_key, dest_bucket,dest_key ):
+def copy_object(source_bucket, source_key, dest_bucket, dest_key, acl=None):
     s3 = boto3.client('s3')
-    s3.copy_object(
-        Bucket=dest_bucket, 
-        CopySource={'Bucket': source_bucket, 'Key':source_key},
-        Key=dest_key
-    )
+
+    params = {
+        'Bucket': dest_bucket,
+        'CopySource': {'Bucket': source_bucket, 'Key': source_key},
+        'Key': dest_key,
+    }
+
+    if acl and acl in valid_acl:
+        params['ACL'] = acl
+
+    s3.copy_object(**params)
 
 def delete_objects(bucket, prefix, objects):
     s3 = boto3.client('s3')
     objects = {'Objects': [{'Key': prefix + o} for o in objects]}
     s3.delete_objects(Bucket=bucket, Delete=objects)
 
 def lambda_handler(event, context):
-    quarantine_bucket = os.environ['QUARANTINEBUCKET']
-    promote_bucket = os.environ['PROMOTEBUCKET']
+    acl = os.environ.get('ACL')
+
+    quarantine_bucket = os.environ.get('QUARANTINEBUCKET')
+    promote_bucket = os.environ.get('PROMOTEBUCKET')
+
+    promote_mode = os.environ.get('PROMOTEMODE', 'move').lower()
+    quarantine_mode = os.environ.get('QUARANTINEMODE', 'move').lower()
+
+    promote_mode = promote_mode if promote_mode in modes else default_mode
+    quarantine_mode = quarantine_mode if quarantine_mode in modes else default_mode
```

Comment:
How about hiding the details of `default_mode` by extracting some methods?

```suggestion
    promote_mode = get_promote_mode()
    quarantine_mode = get_quarantine_mode()
```
```python
def get_promote_mode():
    return get_mode_from_env('PROMOTEMODE')

def get_quarantine_mode():
    return get_mode_from_env('QUARANTINEMODE')

def get_mode_from_env(mode_key):
    mode = os.environ.get(mode_key, 'move').lower()
    return mode if mode in modes else default_mode
```